bigquery/inserter_test.go

```diff
@@ -138,10 +144,29 @@ func TestHandleInsertErrors(t *testing.T) {
 				RowInsertionError{InsertID: "b", RowIndex: 1, Errors: []error{&Error{Message: "m1"}}},
 			},
 		},
+		{
+			description: "invalid index",
+			in: []*bq.TableDataInsertAllResponseInsertErrors{
+				{Errors: []*bq.ErrorProto{{Message: "m0"}}, Index: 2},
+			},
+			want: fmt.Errorf("internal error: unexpected row index: 2"),
+		},
 	} {
 		got := handleInsertErrors(test.in, rows)
-		if !testutil.Equal(got, test.want) {
-			t.Errorf("%#v:\ngot\n%#v\nwant\n%#v", test.in, got, test.want)
+		_, ok := got.(PutMultiError)
+		if ok {
+			// compare structure of the PutMultiError
+			if !testutil.Equal(got, test.want) {
+				t.Errorf("(case: %s)\nin %#v\ngot\n%#v\nwant\n%#v", test.description, test.in, got, test.want)
+			}
+		} else {
```

Comment:
```suggestion
			}
			continue
		}
```

Reply:
Unsure I'm following.  We still want to probe errors when not a PutMultiError, not simply move to the next case.

Reply:
Revisited.  Continue inside the type assertion case, and then do the secondary check.